assets/src/components/vehicleIcon.tsx

```diff
@@ -1,16 +1,299 @@
+  let left = 0
+  let right = 0
+  let top = 0
+  let bottom = 0
+  switch (orientation) {
+    case Orientation.Up:
+      left = -scale * X_CENTER_TO_SIDE
+      top = -scale * Y_CENTER_TO_POINT
+      right = scale * X_CENTER_TO_SIDE
+      bottom = scale * Y_CENTER_TO_BASE + labelBgHeight
+      break
+    case Orientation.Down:
+      left = -scale * X_CENTER_TO_SIDE
+      top = -scale * Y_CENTER_TO_BASE - labelBgHeight
+      right = scale * X_CENTER_TO_SIDE
+      bottom = scale * Y_CENTER_TO_POINT
+      break
+    case Orientation.Left:
+      left = -scale * Y_CENTER_TO_POINT
+      top = -scale * X_CENTER_TO_SIDE
+      right = scale * Y_CENTER_TO_BASE
+      bottom = scale * X_CENTER_TO_SIDE + labelBgHeight
+      break
+    case Orientation.Right:
+      left = -scale * Y_CENTER_TO_BASE
+      top = -scale * X_CENTER_TO_SIDE
+      right = scale * Y_CENTER_TO_POINT
+      bottom = scale * X_CENTER_TO_SIDE + labelBgHeight
+      break
+  }
+  left = Math.min(left, -labelBgWidth / 2)
+  right = Math.max(right, labelBgWidth / 2)
+  const width = right - left
+  const height = bottom - top
 
   return (
-    <g className="m-vehicle-icon" transform={`scale(${scaleAmount})`}>
-      <path
-        className={"m-vehicle-icon__triangle"}
-        d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
+    <svg
+      style={{ width, height }}
+      viewBox={`${left} ${top} ${width} ${height}`}
+    >
+      <VehicleIconSvg
+        size={size}
+        orientation={orientation}
+        label={label}
+        variant={variant}
       />
+    </svg>
+  )
+}
+
+export const VehicleIconSvg = ({
+  size,
+  orientation,
+  label,
+  variant,
+}: Props) => {
+  return (
+    <g className={`m-vehicle-icon m-vehicle-icon${sizeClassSuffix(size)}`}>
+      {label ? (
+        <Label size={size} orientation={orientation} label={label} />
+      ) : null}
+      <Triangle size={size} orientation={orientation} />
+      {variant && variant !== "_" ? (
+        <Variant size={size} orientation={orientation} variant={variant} />
+      ) : null}
     </g>
   )
 }
 
+const Triangle = ({
+  size,
+  orientation,
+}: {
+  size: Size
+  orientation: Orientation
+}) => {
+  const scale = scaleForSize(size)
+  const rotation = rotationForOrientation(orientation)
+  return (
+    <path
+      className="m-vehicle-icon__triangle"
+      d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
+      // Move the center to 0,0
+      transform={`scale(${scale}) rotate(${rotation}) translate(-24,-22)`}
+    />
+  )
+}
+
+const Label = ({
+  size,
+  orientation,
+  label,
+}: {
+  size: Size
+  orientation: Orientation
+  label: string
+}) => {
+  const scale = scaleForSize(size)
```

Comment:
There's a whole lot of disperse logic around each of these components sizing themselves differently. Do you think it might be possible to compose the entire component at a base size, and then scale that once?

Reply:
Ooh, I like that idea. It'd be much easier.

I'm a little worried that the text sizing won't look good if it's just scaled. E.g. at smaller font sizes, you want the variant letter to take up a bigger fraction of the triangle to be readable, but at big sizes you can have spacious margins around the variant letter.

I'll give it a shot and see how it looks.

Reply:
<img width="606" alt="Screen Shot 2019-06-11 at 15 47 57" src="https://user-images.githubusercontent.com/23065557/59301737-4db57f00-8c60-11e9-98f1-2454a89d6d42.png">

That's a lot better than I expected.

Reply:
The original for comparison:
<img width="607" alt="Screen Shot 2019-06-10 at 14 49 25" src="https://user-images.githubusercontent.com/23065557/59222899-99045a80-8b98-11e9-9cea-5f4d5af18ac3.png">

I'm leaning towards no, because the label for the medium size is too big for the ladder and the label for the small ones is too small (we'll use it for turning vehicles)

@ingridpierre what do you think?

Reply:
Here it is with custom text sizes for each size and the big text unbolded.

<img width="614" alt="Screen Shot 2019-06-12 at 10 17 28" src="https://user-images.githubusercontent.com/23065557/59362271-2d89cc80-8d01-11e9-9042-b7bf06a0ab85.png">

After talking with Ingrid and trying it out on a tablet, the tiny font sizes weren't quite readable with the scaling, so I'll stay with the manually specified sizes.